File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>

using namespace std;

int solution(vector<int> nums)
{
    int answer = nums.size() / 2;
    unordered_set<int> unSet;
    
    for(auto num : nums) {
        unSet.insert(num);
    }
    
    if(answer > unSet.size()) {
        answer = unSet.size();
    }
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] nums) {
        int max = nums.length / 2;
        HashSet<Integer> hashSet = new HashSet<>();
        
        for (int num : nums) {
            hashSet.add(num);
        }
        
        if(hashSet.size() > max)
            return max;
        else
            return hashSet.size();
    }
}